unit-tests/types/test-profile-eq.cpp

```diff
@@ -0,0 +1,51 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include "../func/func-common.h"
+#include "../../include/librealsense2/hpp/rs_frame.hpp"
+#include <iostream>
+
+using namespace rs2;
+
+TEST_CASE("test video_stream_profile operator==", "[live]")
+{
+    // Test that for 2 video_stream_profile objects, if width and height are different
+    // then, video_stream_profile.operator== returns false.
+
+    auto devices = find_devices_by_product_line_or_exit(RS2_PRODUCT_LINE_ANY_INTEL);
+    if (devices.size() == 0) return;
+    auto dev = devices[0];
+
+    auto depth_sens = dev.first< rs2::depth_sensor >();
```

Comment:
Add check for cast and exist gracefully if casting fails, i.e. with T265.
Alternatively, modify the flow to handle T265 as well